lib/google/cloud/env/compute_metadata.rb

```diff
@@ -0,0 +1,808 @@
+# frozen_string_literal: true
+
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "faraday"
+require "json"
+
+require "google/cloud/env/compute_smbios"
+require "google/cloud/env/lazy_value"
+require "google/cloud/env/variables"
+
+module Google
+  module Cloud
+    class Env
+      ##
+      # A client for the Google metadata service.
+      #
+      class ComputeMetadata
+        ##
+        # The default host for the metadata server
+        # @return [String]
+        #
+        DEFAULT_HOST = "http://169.254.169.254"
```

Comment:
Why do we use IP instead of host_name ?

Reply:
To remove DNS as a source of failures and latency.